### ISSUE TITLE
Make it more bloated with Kitty image Protocol

### ISSUE THREAD

**Issue by adetabrani** [veltza/st-sx#1]
Hi, some time ago I was exploring reddit and found a fork of ST that implements the kitty image protocol [link](https://github.com/sergei-grechanik/st-graphics) do you have plans to add this feature to your ST? 

disclaimer: I'm not good at programming, especially with C, I'm just curious whether this feature can be implemented together with Sixel, since I discovered this ST fork (very nearly perfect IMO, especially with reflow & sixel)

feel free to close or delete this issue if not necessary 😂, thanks

**Comment by veltza** [comment]
I took a quick look at it and according to this [commit](https://github.com/sergei-grechanik/st-graphics/commit/4380125f0955da29138f585da6ed159e70a60a05) the patch would add more than 5000 lines of code to the project. It would not only make patching difficult, but also increase the maintenance burden a lot, because it only partially supports the kitty image protocol and still has a lot of development to do. In comparison, the sixel implementation has only about 1100 LOC and is therefore more manageable. And since it meets my needs at the moment, I'm not too keen on adding the kitty image protocol now.

However, I'm going to investigate that patch more and see how difficult it would be to add it to the project and what conflicts it would cause with other patches, especially sixels. And if I get it working, maybe I'll add another branch for it. But I can't promise anything.

Ps. if you really need the kitty image protocol, then kitty is the best option, because as far as I know, no other terminal supports all the features of the protocol at the moment.

**Comment by adetabrani** [comment]
Thanks for the feedback, there is no need to rush in implementing it (even if you feel it adds too much complexity then there is no need to implement it), for now the sixel features are enough for me.

I just checked your commit history, especially regarding the "keyboard select" feature, it looks like you are trying to complete a function in vim into it.

have you ever heard of the jump plugin in vim or neovim? like [hop.nvim](https://github.com/phaazon/hop.nvim),
is it possible to implement this feature? in my case my feature is efficient in moving the cursor position to the desired cursor position without having to spend more effort. 

Thanks 😊

**Comment by veltza** [comment]
Yeah, I'm currently rewriting the keyboard select patch for my needs. My goal is to implement only basic vim movements and search. Nothing fancy or complicated.

Still, it would be nice to have [Quick Select Mode](https://wezfurlong.org/wezterm/quickselect.html) like in WezTerm. It is very similar to hop.nvim, but is not used for movement. It just copies text, like links and git hashes, to the clipboard with a few keystrokes. It's very convenient, but I'm afraid it would require a lot of work. So I don't know if I want to start implementing it because it's a nice-to-have, not a must-have feature.

**Comment by adetabrani** [comment]
> have you ever heard of the jump plugin in vim or neovim? like [hop.nvim](https://github.com/phaazon/hop.nvim),
> is it possible to implement this feature? in my case my feature is efficient in moving the cursor position to the desired cursor position without having to spend more effort.


136 / 5.000
I can't believe this feature is finally implemented 😂, maybe in the future it can be implemented for regex hinting features like in alacritty

**Comment by veltza** [comment]
@adetabrani I got your message although it doesn't appear here for some reason. If you're serious about the regex hints, please open a new issue for it. Maybe there is hope for that too.

**Comment by adetabrani** [comment]
actually I want to request that feature, but I think since there is flash mode I think it's enough (although not as efficient as regex hint in copying).

**Comment by adetabrani** [comment]
Looks like @DreamMaoMao already implemented regex hints in his fork

**Comment by veltza** [comment]
@adetabrani It seems so. I haven't had time to look at it yet, so I don't know how complete it is already. Have you tried it yet?

**Comment by adetabrani** [comment]
yes, it's amazing.

maybe, should I close this issue? considering the implementation of kitty protocol seems to be difficult at the moment 🤣

**Comment by DreamMaoMao** [comment]
> yes, it's amazing.
> 
> maybe, should I close this issue? considering the implementation of kitty protocol seems to be difficult at the moment 🤣

Yesterday's version had a few bugs, it should be stable now, you can check my latest commit

**Comment by veltza** [comment]
@adetabrani I think we should close this issue for now and move the discussion about regex hints to this [PR](https://github.com/veltza/st-sx/pull/18).
